SpatialGDK/Source/SpatialGDK/Public/Interop/RPCs/RPC_RingBufferWithACK_Receiver.h

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "EngineClasses/SpatialNetDriverRPC.h"
+#include "Interop/RPCs/RPCTypes.h"
+
+namespace SpatialGDK
+{
+template <typename PayloadType, template <typename> class PayloadWrapper, typename SerializerType>
+class MonotonicRingBufferWithACKReceiver : public TRPCBufferReceiver<PayloadType, PayloadWrapper>
+{
+public:
+	MonotonicRingBufferWithACKReceiver(SerializerType&& InSerializer, int32 InNumberOfSlots, PayloadWrapper<PayloadType>&& InWrapper)
+		: TRPCBufferReceiver<PayloadType, PayloadWrapper>(MoveTemp(InWrapper))
+		, Serializer(MoveTemp(InSerializer))
+		, NumberOfSlots(InNumberOfSlots)
+	{
+		ComponentsToRead.Add(Serializer.GetComponentId());
```

Comment:
Unimportant, but is this genericness useful in other places? This class seems to not rely on `RPCBufferReceiver::OnAdded` for reading components, and might be more concise if it did
```
OnAdded(...)
{
	const ComponentData* RpcComponentData = Element.Components.FindByPredicate(ComponentIdEquality{Serializer.GetComponentId()};
	const ComponentData* AckComponentData = Element.Components.FindByPredicate(ComponentIdEquality{Serializer.GetACKComponentId()};
	OnAdded_ReadAckComponent(AckComponentData);
	OnAdded_ReadRpcComponent(RpcComponentData);
}
```